server/server.js

```diff
@@ -40,30 +40,55 @@ server.listen(PORT, () => {
  * Server endpoints
  */
 
+async function getFiles() {
+  const dataFolderPath = path.join(__dirname, 'data');
+  const files = await fs.promises.readdir(dataFolderPath);
+  return files;
+}
+
 // Endpoint to get a list of files stored on the server
-app.get('/files', (req, res) => {
-  const dataFolderPath = path.join(__dirname, '/data');
-  const fileArray = [];
-  const outputJson = { files: fileArray };
-  fs.readdir(dataFolderPath, (err, files) => {
-    if (err) {
-      console.error(err);
-      res.status(500).send('Internal Server Error!');
-    }
-    if (files.length !== 0) {
-      files.forEach(file => {
-        fileArray.push(file);
-      });
-      outputJson.files = fileArray;
-    }
+app.get('/files', async (req, res) => {
+  try {
+    const fileArray = await getFiles();
+    const outputJson = { files: fileArray };
     res.status(200).send(outputJson);
-  });
+  } catch (error) {
+    console.error(error);
+    res.status(500).send('Internal Server Error!');
+  }
+});
+
+// Endpoint to get most recent file
+app.get('/files/recent', async (req, res) => {
+  const files = await getFiles();
+  if (files.length > 0) {
+    let latestFilePath = null;
+    let latestAge = 0;
+    for (let i=0; i<files.length; i++) {
+      const file = files[i];
+      const filePath = path.join(__dirname, 'data', file);
+      const stats = await fs.promises.stat(filePath);
+      const fileAge = stats.mtime;
+
+      if (fileAge > latestAge) {
+        latestAge = fileAge;
+        latestFilePath = filePath;
+      }
+    }
+    res.download(latestFilePath, err => {
+      if (err) {
+        res.status(404).send('File not found');
+      }
+    });
+  } else {
+    res.status(404).send('No log file found');
+  }
 });
 
 // Endpoint to download file from server
 app.get('/files/:filename', (req, res) => {
   const { filename } = req.params;
-  const filepath = path.join(__dirname, `/data/${filename}`);
+  const filepath = path.join(__dirname, `data/${filename}`);
```

Comment:
Shouldn't it be `path.join(__dirname, 'data', filename);`?

Although, yours does work, I just think it would be cleaner.